core/src/test/java/io/confluent/rest/ApiHeadersTest.java

```diff
@@ -0,0 +1,166 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.rest;
+
+import static org.junit.Assert.assertNull;
+
+import java.io.File;
+import java.security.KeyPair;
+import java.security.cert.X509Certificate;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import javax.net.ssl.SSLContext;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Configurable;
+import javax.ws.rs.core.MediaType;
+import org.apache.http.Header;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.apache.http.ssl.SSLContexts;
+import org.apache.kafka.common.config.types.Password;
+import org.apache.kafka.test.TestSslUtils;
+import org.apache.kafka.test.TestSslUtils.CertificateBuilder;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ApiHeadersTest {
+
+
+  private static String httpUri = "http://localhost:8080";
+  private static String httpsUri = "https://localhost:8081";
+
+  private static final String SSL_PASSWORD = "test1234";
+
+  private static String clientKeystoreLocation;
+  private static TestApplication app;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    File trustStore = File.createTempFile("ApiHeadersTest-truststore", ".jks");
+    File clientKeystore = File.createTempFile("ApiHeadersTest-client-keystore", ".jks");
+    File serverKeystore = File.createTempFile("ApiHeadersTest-server-keystore", ".jks");
+
+    clientKeystoreLocation = clientKeystore.getAbsolutePath();
+
+    Map<String, X509Certificate> certs = new HashMap<>();
+    createKeystoreWithCert(clientKeystore, "client", certs);
+    createKeystoreWithCert(serverKeystore, "server", certs);
+    TestSslUtils.createTrustStore(trustStore.getAbsolutePath(), new Password(SSL_PASSWORD), certs);
+
+    Properties props = new Properties();
+    props.put(RestConfig.LISTENERS_CONFIG, httpUri + "," + httpsUri);
+    props.put(RestConfig.SSL_KEYSTORE_LOCATION_CONFIG, serverKeystore.getAbsolutePath());
+    props.put(RestConfig.SSL_KEYSTORE_PASSWORD_CONFIG, SSL_PASSWORD);
+    props.put(RestConfig.SSL_KEY_PASSWORD_CONFIG, SSL_PASSWORD);
+
+    app = new TestApplication(new TestRestConfig(props));
+    app.start();
+  }
+
+  @AfterClass
+  public static void teardown() throws Exception {
+    if (app != null) {
+      app.stop();
+    }
+  }
+
+  private static void createKeystoreWithCert(File file, String alias, Map<String, X509Certificate> certs) throws Exception {
```

Comment:
nit: private statics to the bottom of the file.

Reply:
done